gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/OAuth2Provider.java

```diff
@@ -187,23 +190,27 @@ private void initRouter() {
                 .handler(new AuthorizationRequestValidateParametersHandler(domain))
                 .handler(authenticationFlowHandler.create())
                 .handler(new AuthorizationRequestResolveHandler())
-                .handler(new AuthorizationRequestEndUserConsentHandler(userConsentService, domain))
+                .handler(new AuthorizationRequestEndUserConsentHandler(userConsentService))
                 .handler(new AuthorizationEndpoint(flow))
-                .failureHandler(new AuthorizationRequestFailureHandler(domain, openIDDiscoveryService, jwtService, jweService));
+                .failureHandler(new AuthorizationRequestFailureHandler(openIDDiscoveryService, jwtService, jweService));
 
         // Authorization consent endpoint
         Handler<RoutingContext> userConsentPrepareContextHandler = new UserConsentPrepareContextHandler(clientSyncService);
         oauth2Router.route(HttpMethod.GET, "/consent")
+                .handler(new AuthorizationRequestParseClientHandler(clientSyncService))
```

Comment:
`AuthorizationRequestParseClientHandler`  and `UserConsentPrepareContextHandler` seems to do the same thing (at least get the client).

Reply:
You're right. I think it makes sense to just remove the `fetchClient` invokation in `UserConsentPrepareContextHandler` as it is handled by `AuthorizationRequestParseClientHandler`.